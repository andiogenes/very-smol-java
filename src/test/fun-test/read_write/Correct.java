class Main {
    class Foo {
        // Поле вложенного класа
        long bar;
    }

    // Поля класса
    long ba = 1000;
    long boo = ba;

    void main() {
        println(boo);
        println();

        long foo;
        int bar;
        short baz;

        // Составной оператор
        {
            long foo;
            int bar;
            short baz;
            foo = bar = baz = 25;
            println(foo);
            println(bar);
            println(baz);
            println();
        }

        // Вне составного оператора
        println(foo);
        println(bar);
        println(baz);
        println();

        // Цепочка присваиваний
        boo = Main.ba = foo = bar = baz = 10;
        println(foo);
        println(bar);
        println(baz);
        println(boo);
        println(ba);
        println();

        // Присваивание по отдельности
        foo = 10;
        bar = 20;
        baz = 30;
        boo = 40;
        Main.ba = 50;

        println(foo);
        println(bar);
        println(baz);
        println(boo);
        println(ba);
        println();

        // Присваивание полю вложенного класса
        Main.Foo.bar = foo;
        println(Main.Foo.bar);
    }
}