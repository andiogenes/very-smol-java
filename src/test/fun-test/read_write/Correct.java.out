1000

25
25
25

0
0
0

10
10
10
10
10

10
20
30
40
50

10
